commit f85758a3b432e9b4b4e83ffe85bbc3684b985e3e
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2019-03-06T18:39:36+00:00 (7 months ago)
    
    TRITON-1287 agentsshar no longer able to assemble agent branches
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/Makefile b/Makefile
index 8c52a3b..0cd0154 100644
--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ TOP ?= $(error Unable to access eng.git submodule Makefiles.)
 NAME = agentsshar
 
 ifeq ($(BUILDNAME),)
-    BUILDNAME=master
+    BUILDNAME=$(BRANCH)
 endif
 
 CLEAN_FILES += build/agents
@@ -26,7 +26,7 @@ deps:
 	PATH=/opt/tools/bin:$(PATH) /opt/tools/bin/npm install
 
 shar: deps
-	PATH=/opt/tools/bin:$(PATH) ./mk-agents-shar -b $(BUILDNAME)
+	PATH=/opt/tools/bin:$(PATH) ./mk-agents-shar -b "$(BUILDNAME)"
 
 publish: shar
 	mkdir -p $(ENGBLD_BITS_DIR)/$(NAME)
diff --git a/mk-agents-shar b/mk-agents-shar
index de22349..c3e86d7 100755
--- a/mk-agents-shar
+++ b/mk-agents-shar
@@ -18,9 +18,6 @@
 #    # Source dir (-d) is a local path or Manta path.
 #    bin/mk-agents-shar -b release-20110714 -d /Joyent_Dev/public/builds
 #
-#    # Default build name is "master", default source dir is as above.
-#    bin/mk-agents-shar
-#
 #    # Using built agent packages from a local "bits" dir.
 #    bin/mk-agents-shar -b release-20110714 -d ../../bits
 #
@@ -67,7 +64,6 @@ if [[ -n "$ENGBLD_DEST_OUT_PATH" ]]; then
 else
 	DEFAULT_SOURCE_DIR=${MANTA_BASE}
 fi
-DEFAULT_BUILD_NAMES=master
 ALL_AGENTS=$(cat agents.json | json -a | xargs)
 
 BUILD_DIR=build
@@ -100,7 +96,7 @@ function usage() {
     echo ""
     echo "Options:"
     echo "  -h      print this help and exit"
-    echo "  -b      build name(s), e.g. 'master' (the default), 'release-20110714'"
+    echo "  -b      build name(s), e.g. 'master', 'release-20110714'"
     echo "          Typically the name of the repo branch. Note this can"
     echo "          be multiple space-separated build names (e.g. "
     echo "          'feature1 master'). Each name is attempted to find an "
@@ -313,7 +309,6 @@ EOF
 #---- mainline
 
 # Process arguments.
-build_names=$DEFAULT_BUILD_NAMES
 source_dir=$DEFAULT_SOURCE_DIR
 output_dir=$BUILD_DIR
 while getopts "hb:d:o:" c; do
@@ -339,6 +334,12 @@ while getopts "hb:d:o:" c; do
 done
 shift $((OPTIND - 1))
 
+if [[ -z "$build_names" ]]; then
+    echo "ERROR: build_names (-b) is required"
+    usage
+    exit 1
+fi
+
 agents="$*"
 if [[ -z "$agents" ]]; then
     agents="$ALL_AGENTS"
